### DISCUSSION TITLE
I can't get the latest/new `num` state in `useEffect(, [])`, but `Store.getState().num` does.

### DISCUSSION THREAD

**Discussion by SolidZORO** [pmndrs/zustand#1834] [Q&A] [answered]
I can't get the latest/new `num` state in `useEffect(, [])`, but `Store.getState().num` does. 

I know this is not a `zustand` problem, is there any way to avoid this error? `eslint plugin` or some other method?

https://codesandbox.io/s/beautiful-sound-65ppgh?file=/src/App.tsx

<img width="900" alt="image" src="https://github.com/pmndrs/zustand/assets/661587/34f3baff-70a6-4970-981b-2e2015ad8d3f">

**Comment by dbritto-dev** [comment, marked as answer]
@SolidZORO you can use `subscribe` to see previous and current values. BTW, with `useEffect` + num as dep you can get any current/latest value
